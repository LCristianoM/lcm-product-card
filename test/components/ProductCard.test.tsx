import React from 'react';
import TestRenderer from 'react-test-renderer';
import { product2 } from '../data/products';
import { ProductCard } from '../../src/components/ProductCard';

const { act } = TestRenderer;


describe('ProductCard', () => {
  test('debe mostrar el componete correctamene ', () => {
    const wrapper = TestRenderer.create(
      <ProductCard product={product2}>
        {() => <h1>Â¨Product card</h1>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test(' debe incrementat el contador', () => {
    const wrapper = TestRenderer.create(
      <ProductCard product={product2}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product card</h1>
            <span> {count} </span>
            <button onClick={() => increaseBy(2)}></button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick ();
    })

    tree = wrapper.toJSON();

    expect( ( tree as any).children[1].children[0]).toBe('1');

  });
});
