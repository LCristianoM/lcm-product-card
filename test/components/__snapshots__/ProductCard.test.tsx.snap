// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard  debe incrementat el contador 1`] = `
<div
  className="productCard undefined "
>
  <h1>
    Product card
  </h1>
  <span>
     
    0
     
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard debe mostrar el componete correctamene  1`] = `
<div
  className="productCard undefined "
>
  <h1>
    Â¨Product card
  </h1>
</div>
`;
